### PR TITLE
QW-885: Move to AWS SDK

### PR DESCRIPTION
**Pull request by ChillFish8**
Moves from the now unmaintained Rusto library to the official AWS SDK.

### TODO List

- [x] Migrate `quickwit-aws` crate
- [x] Move across S3 handling
- [x] Fix Kinesis 
- [x] Fix unit tests
- [x] Check semantics with other providers (Google storage, B2, MinIO, is the same as before)
- [x] Replug the different authentication methods
- [x] Update documentation in https://quickwit.io/docs/guides/aws-setup#aws-credentials
- [x] Test virtual path on s3

### CODE REVIEW

File: quickwit/quickwit-aws/src/lib.rs [hunk 1]
```diff
@@ -19,94 +19,55 @@
...
-    fn new() -> Self {
-        let mut chain_provider = ChainProvider::new();
-        chain_provider.set_timeout(CREDENTIALS_FETCH_TIMEOUT);
-        let web_identity_provider = WebIdentityProvider::from_k8s_env();
-        Self {
-            chain_provider,
-            web_identity_provider,
-        }
-    }
+            aws_config::from_env()
```
Comment (fmassot) [inline quickwit/quickwit-aws/src/lib.rs:60]: Did you check if it covers the k8s case? (it seems so but just want you to double check). We could add a task such as "test k8s credentials on k8s cluster" in the to-do list.

**Comment by fmassot** [comment]
Sounds like a good start. One thing you can add to the to-do list is to test virtual-hosted–style requests.

File: quickwit/quickwit-storage/src/error.rs [hunk 2]
```diff
@@ -47,6 +49,10 @@ pub enum StorageResolverError {
     /// A protocol is required for the URI.
     #[error("Invalid format for URI: required: `{message}`")]
     InvalidUri { message: String },
+    #[error("The system does not have a initialised AWS config available")]
+    /// This should never happen if the `quickwit_aws::try_init_aws_config` method
+    /// has been called, if this error occurs this is a bug.
+    MissingAWSConfig,
```
Comment (fmassot) [inline quickwit/quickwit-storage/src/error.rs:55]: Can we rename `MissingAWSConfig`? I think it concerns any s3 compatible storage. 
And I'm a bit confused here, the issue is not really that the config is missing right? I think we should be clearer and say something like "S3 compatible storage configuration has not been initialized. This should never happen".

File: quickwit/quickwit-aws/src/lib.rs (hunk 1 above)
Comment (ChillFish8) [reply quickwit/quickwit-aws/src/lib.rs:60]: I haven't checked, what system does it normally use? (I.e. ENV vars, etc...)

Comment (fmassot) [reply quickwit/quickwit-aws/src/lib.rs:60]: I think you work on it in the PR related to credentials errors (web identity provider etc.). But the defaults seem quite nice so hopefully we don't have much to do there.

**Comment by ChillFish8** [comment]
Marking as ready just to try get CI running

**Comment by fulmicoton** [comment]
@fmassot can we land this?

File: quickwit/quickwit-storage/src/error.rs (hunk 2 above)
Comment (guilload) [reply quickwit/quickwit-storage/src/error.rs]: I got rid of it altogether.

**Review by fmassot — APPROVED** [review]

**Pull request merged**